core/node/events/snapshot.go

```diff
@@ -607,11 +607,34 @@ func update_Snapshot_Member(
 		default:
 			return RiverError(Err_INVALID_ARGUMENT, fmt.Sprintf("unknown member blockchain transaction type %T", transactionContent))
 		}
+	case *MemberPayload_Mls_:
+		return update_Snapshot_Mls(iSnapshot, content.Mls)
 	default:
 		return RiverError(Err_INVALID_ARGUMENT, fmt.Sprintf("unknown membership payload type %T", memberPayload.Content))
 	}
 }
 
+func update_Snapshot_Mls(
+	iSnapshot *Snapshot,
+	mlsPayload *MemberPayload_Mls,
+) error {
+	if iSnapshot.Members.GetMls() == nil {
+		iSnapshot.Members.Mls = &MemberPayload_Snapshot_Mls{}
+	}
+	snapshot := iSnapshot.Members.Mls
+	switch content := mlsPayload.Content.(type) {
+	case *MemberPayload_Mls_InitializeGroup_:
+		if len(snapshot.ExternalGroupSnapshot) > 0 || len(snapshot.GroupInfoMessage) > 0 {
+			return RiverError(Err_INVALID_ARGUMENT, "duplicate mls initialization")
+		}
+		snapshot.ExternalGroupSnapshot = content.InitializeGroup.ExternalGroupSnapshot
+		snapshot.GroupInfoMessage = content.InitializeGroup.GroupInfoMessage
+		return nil
```

Comment:
we need to copy these instead of just assigning, otherwise we get data races 
snapshot.ExternalGroupSnapshot = content.InitializeGroup.ExternalGroupSnapshot
		snapshot.GroupInfoMessage = content.InitializeGroup.GroupInfoMessage

Reply:
I had this problem with the key solicitations, but i needed to clone the object, event := proto.Clone(keySolicitation).(*MemberPayload_KeySolicitation)
here because we're just moving slices, maybe we don't? worth looking into